cajita/src/Cajita_BovWriter.hpp

```diff
@@ -171,29 +211,45 @@ void writeTimeStep( const int time_step_index, const double time,
                 owned_extents[d] = owned_index_space.extent( d ) + 1;
         }
     }
-    owned_extents[3] = array.layout()->dofsPerEntity();
+    owned_extents[num_space_dim] = array.layout()->dofsPerEntity();
+
+    // Gather halo data if any dimensions are periodic.
+    for ( std::size_t d = 0; d < num_space_dim; ++d )
+    {
+        if ( global_grid.isPeriodic( d ) )
+        {
+            auto halo =
+                createHalo( NodeHaloPattern<num_space_dim>(), 0, array );
+            halo->gather( execution_space(), array );
+            break;
+        }
+    }
 
     // Create a contiguous array of the owned array values. Note that we
     // reorder to KJI grid ordering to conform to the BOV format.
-    IndexSpace<4> local_space(
-        { owned_index_space.min( Dim::I ), owned_index_space.min( Dim::J ),
-          owned_index_space.min( Dim::K ), 0 },
-        { owned_index_space.min( Dim::I ) + owned_extents[Dim::I],
-          owned_index_space.min( Dim::J ) + owned_extents[Dim::J],
-          owned_index_space.min( Dim::K ) + owned_extents[Dim::K],
-          owned_extents[3] } );
+    std::array<long, num_space_dim + 1> local_space_min;
+    std::array<long, num_space_dim + 1> local_space_max;
+    for ( std::size_t d = 0; d < num_space_dim; ++d )
+    {
+        local_space_min[d] = owned_index_space.min( d );
+        local_space_max[d] = owned_index_space.min( d ) + owned_extents[d];
+    }
+    local_space_min.back() = 0;
+    local_space_max.back() = owned_extents.back();
+    IndexSpace<num_space_dim + 1> local_space( local_space_min,
+                                               local_space_max );
     auto owned_subview = createSubview( array.view(), local_space );
-    IndexSpace<4> reorder_space( { owned_extents[Dim::K], owned_extents[Dim::J],
-                                   owned_extents[Dim::I], owned_extents[3] } );
+
+    std::array<long, num_space_dim + 1> reorder_space_size;
```

Comment:
Should this just be inside `reorderView`?

Reply:
It could be but `reorderView` was really added to deal with the differences in 2d/3d indexing.